src/components/DataViewer/DataViewer.tsx

```diff
@@ -1,78 +1,64 @@
-import CloseIcon from "@mui/icons-material/Close";
+"use client";
+
 import React, { ReactElement } from "react";
-import Modal from "react-modal";
 import styled from "styled-components";
+import Modal from "@/components/Modal/Modal";
+
+type valueType = string[] | string | number | boolean | null;
 
 export interface Data {
-    [key: string]: string | number | null;
+    [key: string]: valueType;
 }
 
-const StyledModal = styled(Modal)`
-    display: flex;
-    flex-direction: column;
-    position: absolute;
-    top: 50%;
-    left: 50%;
-    transform: translate(-50%, -50%);
-    border: 0;
-    border-radius: 5px;
-    outline: none;
-    padding: 0;
-    width: 80%;
-    max-height: 80%;
-    background: rgb(255, 255, 255);
-    box-shadow: rgb(200, 200, 200) 0 2px 10px;
-    overflow: hidden;
-`;
-const Header = styled.h3`
-    display: flex;
-    justify-content: space-between;
-    align-items: center;
-    background-color: #eeeeee;
-    padding: 0.5rem;
-    margin-top: 0;
-`;
-const Content = styled.div`
-    overflow: scroll;
-    margin: 1.5rem;
-    margin-top: 1rem;
-`;
 const Key = styled.div`
-    color: grey;
-    font-size: small;
+    // TODO VFB-16 Add a dark font colour (black perhaps) that is accessible with the colour below
+    color: ${(props) => props.theme.secondaryForegroundColor};
+
+    // TODO VFB-16 Add the equivalent of this colour (secondaryBackgroundColor}40) to a palette without the transparency
+    background-color: ${(props) => props.theme.secondaryBackgroundColor};
+
+    display: inline-block;
+    border-radius: 0.7em;
+    padding: 0.2em 0.5em;
 `;
+
 const Value = styled.div`
-    font-size: medium;
+    padding: 0.2em 0.5em;
 `;
+
 const EachItem = styled.div`
-    padding-bottom: 1rem;
+    padding-bottom: 1em;
 `;
-const CloseButton = styled.button`
-    border: 0;
-    background-color: red;
-    color: white;
-    display: flex;
-    justify-content: center;
+
+const ContentDiv = styled.div`
+    width: 1000px;
+    max-width: 100%;
 `;
 
-interface ClearButtonProps {
-    closeModal: () => void;
-}
+const valueIsEmpty = (value: valueType): boolean => {
+    return (Array.isArray(value) && value.length === 0) || value === "" || value === null;
+};
 
-const ClearButton: React.FC<ClearButtonProps> = (props) => {
-    return (
-        <CloseButton onClick={props.closeModal}>
-            <CloseIcon />
-        </CloseButton>
-    );
+const formatDisplayValue = (value: valueType): string => {
+    if (valueIsEmpty(value)) {
+        return "-";
+    }
+
+    if (typeof value === "boolean") {
+        const booleanString = value.toString();
+        return booleanString[0].toUpperCase() + booleanString.slice(1);
+    }
+
+    return value!.toString();
```

Comment:
why is the exclamation mark needed? is there a type which 'gets through' the type system and passes through `valueIsEmpty` such that `value` could be 'empty' at this point? (`undefined`?)

Maybe `valueIsEmpty` is better defined as `(array bit) || !value;`? does this catch too much?

Reply:
Doesn't look like the type system can infer that the return value from `valueIsEmpty` cannot be `null`. That's why we had to add the exclamation mark